index.php

```diff
@@ -18,13 +20,29 @@
     <link rel="stylesheet" href="styles.css">
 </head>
 <body>
+    <nav>
+        <div class="searchContainer">
+            <form>
+                <input type="search" name="query" placeholder="search by title" aria-label="search by title"
+                       role="search" value="<?php if(isset($_GET['query'])) {echo $_GET['query'];} ?>">
+                <button type="submit">Search</button>
+            </form>
+            <a href="index.php"><button>Clear</button></a>
+        </div>
+    </nav>
     <main id="homePage">
         <h1>Mongoose Game Listing</h1>
         <div class="cardContainer">
             <?php
+            if (isset($_GET['query']) && !empty($_GET['query'])) {
+                foreach ($gamesByQuery as $game) {
+                    echo GameViewHelper::createGameCard($game);
+                }
+            } else {
                 foreach($games as $game) {
                     echo GameViewHelper::createGameCard($game);
                 }
+            }
```

Comment:
code duplication - instead of having the same block of code twice, you can use the same variable $games and if there is a GET query it can have the filtered results instead of all the results

Reply:
Resolved